client/src/pages/JobDetails.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import { fetchOneJob, Job } from '../lib';
+import Footer from '../components/Footer';
+import { useParams } from 'react-router-dom';
+import JobInfoStack from '../components/JobInfoStack';
+import ReactMarkdown from 'react-markdown';
+
+export default function JobDetails() {
+  const { jobId } = useParams();
+  const [job, setJob] = useState<Job>();
+  const [loading, setLoading] = useState<boolean>();
+
+  useEffect(() => {
+    async function loadOneJob(jobId: number) {
+      setLoading(true);
+      try {
+        const job = await fetchOneJob(jobId);
+        setJob(job);
+      } catch (err) {
+        console.error(err);
+      } finally {
+        setLoading(false);
+      }
+    }
+    if (jobId) {
+      loadOneJob(+jobId);
+    }
+  }, [jobId]);
+
+  if (loading) {
+    return (
+      <div className="font-satoshi bg-stone-200">
```

Comment:
You probably display this Loading message in multiple places (or will before you're done). This would be a good thing to put into a custom component.